pkg/hub/submarineragent/controller.go

```diff
@@ -497,6 +515,28 @@ func (c *submarinerAgentController) updateSubmarinerConfigStatus(ctx context.Con
 	return err
 }
 
+func (c *submarinerAgentController) updateManagedClusterAddOnStatus(ctx context.Context,
+	managedClusterAddon *addonv1alpha1.ManagedClusterAddOn, brokerNamespace string) error {
+	message := fmt.Sprintf("Waiting for brokers.submariner.io object named submariner to be created in %q namespace", brokerNamespace)
+	condition := metav1.Condition{
+		Type:    BrokerCfgMissing,
+		Status:  metav1.ConditionFalse,
+		Reason:  "SubmarinerBrokerConfigMissing",
+		Message: message,
+	}
```

Comment:
The `Status` should be `metav1.ConditionTrue` since we're asserting that the broker config is missing. Also, we should assert the opposite `Status` once the broker config is available.

However, I think the condition should reflect the status of the globalnet config. So define the type as "_GlobalnetConfigApplied_". In this case, it would have `metav1.ConditionFalse` with `Reason` "_BrokerConfigMissing_". When the globalnet config is applied, then update `Status`  to `metav1.ConditionTrue`.

Reply:
But this condition only indicates that it's missing w/o indicating any ramifications. I think it makes more sense to assert a condition indicating what you're trying to actually do, in this case apply the globalnet config. A missing Broker object is one reason the condition isn't satisfied - if an error occurs we could also indicate that in the reason/message. Also when it does get successfully applied, the message can include the info that was applied, ie the CIDR etc.

In the future, if we apply other config from the Broker object, we can assert separate conditions appropriately ("_XYZApplied_").

Reply:
@tpantelis is it okay to address it in a separate PR and get the current one merged?